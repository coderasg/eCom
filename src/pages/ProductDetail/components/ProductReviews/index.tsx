import React from 'react';

import BookMarkFlag from './../../../../components/Icons/BookMarkFlag/index.svg';

import CommentCount from './../../../../components/CommentCount';
import Like from './../../../../components/Like';

import { $iconDefaultColor } from './../../../../components/Icons/IconColors';

import styles from './productReviews.module.scss';

type Props = {
  likeCount: number;
  commentCount: number;
};

function ProductReviews(props: Props) {
  const { likeCount, commentCount } = props;

  return (
    <div data-testid="ProductReviews-wrapper" className={styles.productReviews}>
      <section className={styles.reviews}>
        <Like count={likeCount} />
        <CommentCount count={commentCount} />
      </section>
      <section className={styles.reportFlag}>
        <BookMarkFlag width={25} height={25} fill={$iconDefaultColor} />
      </section>
    </div>
  );
}

export default ProductReviews;
