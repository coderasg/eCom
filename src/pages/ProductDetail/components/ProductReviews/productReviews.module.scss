.productReviews {
  display: flex;
  align-items: center;
  justify-content: space-between;
  padding: 10px;

  .reviews {
      display: flex;
      justify-content: space-around;
      min-width: 100px;
  }
}
