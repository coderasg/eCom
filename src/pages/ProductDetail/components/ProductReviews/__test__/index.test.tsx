import React from 'react';
import renderer from 'react-test-renderer';

import ProductReviews from '../index';

describe('(Component) ProductReviews', () => {
  const defaultProps = {
    likeCount: 11,
    commentCount: 11
  };
  
  test('should pass snapshot test', () => {
    const wrapper = renderer.create(<ProductReviews {...defaultProps} />);
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
