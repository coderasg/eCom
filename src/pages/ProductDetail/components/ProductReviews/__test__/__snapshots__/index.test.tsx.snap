// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) ProductReviews should pass snapshot test 1`] = `
<div
  data-testid="ProductReviews-wrapper"
>
  <section>
    <span
      data-testid="Like-wrapper"
    >
      <SvgIcon
        height={15}
        width={15}
      />
      <span>
        11
      </span>
    </span>
    <div
      data-testid="CommentCount-wrapper"
    >
      <SvgIcon
        fill="#b1b0b5"
        height={25}
        width={25}
      />
      11
    </div>
  </section>
  <section>
    <SvgIcon
      fill="#b1b0b5"
      height={25}
      width={25}
    />
  </section>
</div>
`;
